linked_list/.~c9_invoke_17vQM9.py

```diff
@@ -0,0 +1,172 @@
+class AbstractLinkedList(object):
+    """
+    Abstract class representing the LinkedList inteface you must respect.
+    
+    You must not implement any of the method in this class, and this class
+    must never be instantiated. It's just a "guide" of which methods
+    the LinkedList class should respect.
+    """
+
+    def __str__(self):
+        raise NotImplementedError()
+
+    def __len__(self):
+        raise NotImplementedError()
+
+    def __iter__(self):
+        raise NotImplementedError()
+
+    def __getitem__(self, index):
+        raise NotImplementedError()
+
+    def __add__(self, other):
+        raise NotImplementedError()
+
+    def __iadd__(self, other):
+        raise NotImplementedError()
+
+    def __eq__(self, other):
+        raise NotImplementedError()
+
+    def append(self, element):
+        raise NotImplementedError()
+
+    def count(self):
+        raise NotImplementedError()
+
+    def pop(self, index=None):
+        raise NotImplementedError()
+
+
+class Node(object):
+    """
+    Node class representing each of the linked nodes in the list.
+    """
+
+    def __init__(self, elem, next=None):
+        self.elem = elem
+        self.next = next
+
+    def __str__(self):
+        return str(self.elem)
+
+    def __eq__(self, other):
+        if isinstance(other, LinkedList):
+            return self.elem == other.elem #compare node.elem to value passed to it
+        else:
+            return self.elem == other
+
```

Comment:
you need to check if other is an instance of Node not LinkedList. self is a Node so you need to make sure other is also a node. 

also you need to check if self.next is equal to other.next. for example if you have a node where elem is 3 and next points to another node and another node where elem is 3 and next points to nothing, you would return True when they are not equal